framework/src/main/java/org/apache/drill/test/framework/DrillConfigRenderer.java

```diff
@@ -0,0 +1,152 @@
+
+        StringBuilder sb = new StringBuilder(prefixIndent)
+                .append(configKey)
+                .append(":");
+
+        //String should be within "double quotes".
+        if (configVal instanceof String) {
+            sb.append("\"")
+                    .append(configVal)
+                    .append("\"");
+        } else if (configVal instanceof DrillConfigRenderer) {
+            //If DrillConfigRenderer type, call render.
+            sb.append(((DrillConfigRenderer) configVal).render(acc));
+        } else {
+            sb.append(configVal);
+        }
+
+        sb.append(",");
+        return sb.append("\n").toString();
+    }
+
+    /**
+     * Defines the format to serialize a config key and value to HOCON format
+     * where config value is of type {@link List} .
+     * The method assumes that there are more configurations to follow.
+     * Hence introduces a "comma (,)" at the very end. If it is the last field,
+     * the caller has to ensure to remove it.
+     * @param acc
+     * @param configKey
+     * @param configVal
+     * @param <K>
+     * @param <V>
+     * @return formatted string for a config key-value pair, in HOCON format.
+     */
+    default <K, V> String formatConfig(final int acc,
+                                    final K configKey,
+                                    final List<V> configVal) {
+        Preconditions.checkNotNull(configKey, "config key cannot be null!");
+        Preconditions.checkNotNull(configVal, "config value cannot be null!");
+
+        String prefixIndent = indent(acc); //indentation prefix
+
+        StringBuilder sb = new StringBuilder(prefixIndent)
+                .append(configKey)
+                .append(":")
+                .append("["); //begin square brackets for list
+
+        if(configVal.isEmpty()) { //if empty, close and return
+            return sb.append("]").append(",").append("\n").toString();
+        }
+
+        if (configVal.get(0) instanceof String) {
+            configVal.forEach(val -> sb.append("\"")
+                            .append(val)
+                            .append("\"").append(","));
+            sb.deleteCharAt(sb.length()-1).append("]").append(",");
+        } else if (configVal.get(0) instanceof DrillConfigRenderer) {
+            //Handle values of type DrillConfigRenderer separately
+            //Unlike other types, each value will be rendered in a new line.
+            IntStream
+                    .range(0, configVal.size()-1)
+                    .forEach(index ->
+                            sb.append(((DrillConfigRenderer) configVal.get(index)).render(acc+2))
+                                    .append(",")
+                                    .append("\n")
+                                    .append(indent(acc+2)));
+
+            sb.append(((DrillConfigRenderer) configVal.get(configVal.size()-1)).render(acc+2))
+                    .append("\n").append(prefixIndent).append("]").append(",");
```

Comment:
I am not sure if I understood the question.

`formatConfig` assumes that the config is not the last entry, hence it would always add a trailing `,` at the end. If you are referring to a case where the `configVal` is a `List` of values (especially of type `DrillConfigRenderer`), then the trailing `,` in **line 131** refers to a `,` after the list itself and does not refer to a trailing `,` within the list.